--- importer.py.orig	2025-02-06 22:58:22.270770005 +0000
+++ importer.py	2025-02-06 22:58:37.008770012 +0000
@@ -184,9 +184,10 @@
         output_folder: str = None,
         model_name: str = None,
         simulator_type: str = AMICI,
+        base_path: str | Path = None
     ) -> PetabImporter:
         """Simplified constructor using a petab yaml file."""
-        petab_problem = petab.Problem.from_yaml(yaml_config)
+        petab_problem = petab.Problem.from_yaml(yaml_config, base_path)
 
         return PetabImporter(
             petab_problem=petab_problem,
